client-sdk/go/types/transaction.go

```diff
@@ -203,25 +212,39 @@ func (t *Transaction) PrepareForSigning() *TransactionSigner {
 	}
 }
 
-// PrettyType returns a representation of the Transaction that can be used for pretty printing.
-func (t *Transaction) PrettyType(txBody interface{}) (*PrettyTransaction, error) {
-	if err := cbor.Unmarshal(t.Call.Body, txBody); err != nil {
-		return nil, fmt.Errorf("failed to unmarshal transaction call body: %w", err)
+func (t *Transaction) PrettyPrint(ctx context.Context, prefix string, w io.Writer) {
+	pt, err := t.PrettyType()
```

Comment:
how does this end up working, does the caller of PrettyPrint have to ensure that all relevant modules are included in their program for the method names to be registered?

Reply:
Correct. It's the same concept we're using in oasis-core. Once you load the oasis-sdk go module, the `NewMethodName()` is called for each runtime method.